alsoHeresTheInterviewQuestionIHad/alsoHeresTheInterviewQuestionIHad.js

```diff
@@ -0,0 +1,43 @@
+const fs = require("fs");
+const buildings = fs.readFileSync("data.txt", "utf-8").split(",");
+
+function bestArea(streetArray) {
+    let numberOfBuildings = streetArray.length;
+
+    let tallestBuildingSoFar = 0;
+    let tallestBuilding = Math.max(...streetArray);
+    let tallestBuildingRemaining = tallestBuilding;
+
+    let bannerOneHeight = 0;
+    let bannerOneWidth = 0;
+    let bannerOneArea = tallestBuilding * numberOfBuildings;
+    let bannerTwoHeight = 0;
+    let bannerTwoWidth = 0;
+    let bannerTwoArea = 0;
+
+    let bestAreaInFunction = bannerOneArea;
+
+    for (let i = 0; i < numberOfBuildings - 1; i++) {
+        if (streetArray[i] > tallestBuildingSoFar) {
+            tallestBuildingSoFar = streetArray[i];
+        }
+
+        if ((streetArray[i] = tallestBuildingRemaining)) {
+            tallestBuildingRemaining = Math.max(...streetArray.slice(i + 1, streetArray.length));

```

Comment:
Okay, so I added this to appease the lesser gods.  Now it only bothers checking what's the tallest building remaining on the street if the current building is part of the group known as the tallest building remaining.  That way if it IS the only member of tallest building remaining, we'll snag whatever building is tallest from the smaller group.  But from then on, it's not checking them, it just remembers what the tallest building is until it actually encounters one of that height.